### PR TITLE
[12.0] database_cleanup: refactor tests file

### PR DESCRIPTION
**Pull request by maisim**
* database_cleanup: [IMP] refactoring tests file
Split the big test method in tiny ones to ease read of results and to make adding tests easier 

* database_cleanup: [FIX] missing cleanup in tests
Remove tests data in db to avoid fail on next run on the same db

### CODE REVIEW

**Comment by maisim** [comment]
The build fail with 
`  File "/mnt/data/odoo-addons-dir/setup/company_country/odoo/addons/company_country/models/res_config.py", line 39, in load_company_country
    'COUNTRY environment variable with country code is not '
odoo.exceptions.ValidationError: ('COUNTRY environment variable with country code is not set and no localization module is marked to be installed.', None)`

Any idea?

**Comment by maisim** [comment]
Salut  @sbidoul  @gurneyalex  !

An opinion about this https://github.com/OCA/server-tools/issues/2296 ?

**Comment by thomaspaulb** [comment]
@maisim Hi, quite some time ago but could you rebase this? I think it could still be valuable.

**Comment by maisim** [comment]
@thomaspaulb Hi, It's done

File: database_cleanup/tests/test_database_cleanup.py [hunk 1]
```diff
@@ -8,138 +8,139 @@
...
             "where indexname='res_partner_name_index' and "
             "tablename='res_partner'"
         )
         self.assertEqual(self.env.cr.rowcount, 1)
-        # duplicate a property
-        duplicate_property = self.env['ir.property'].search([], limit=1).copy()
-        purge_property = self.env['cleanup.purge.wizard.property'].create({})
+
+    def test_purge_duplicate_property(self):
+        # check if duplicate a property is deleted
```
Comment (ypapouin) [inline database_cleanup/tests/test_database_cleanup.py:39]: check if a duplicated property is deleted

**Review by ypapouin — COMMENTED** [review]
LGTM (code review), thanks for this refactoring.
Please a next time apply the black/yapf format in a first commit to simplify the review.

**Comment by legalsylvain** [comment]
Hi @maisim. Thanks for your contribution.
I think that this refactor should be done in more recent version, ideally in v17.0 https://github.com/OCA/server-tools/tree/17.0/database_cleanup

**Review by alexis-via — APPROVED** [review]

**Comment by thomaspaulb** [comment]
/ocabot merge patch

@maisim If you still have time to create another PR that forward ports this to another version, that would be greatly appreciated.

**Pull request merged**